hack/docs/configuration_gen_docs.go

```diff
@@ -48,11 +48,12 @@ func main() {
 	topDoc := fmt.Sprintf("%s%s\n\n", startDocSections[0], genStart)
 	bottomDoc := fmt.Sprintf("\n%s%s", genEnd, endDocSections[1])
 
-	opts := options.New()
+	fs := flag.NewFlagSet("karpenter", flag.ContinueOnError)
+	(&options.Options{}).AddFlags(fs)
 
 	envVarsBlock := "| Environment Variable | CLI Flag | Description |\n"
 	envVarsBlock += "|--|--|--|\n"
-	opts.FlagSet.VisitAll(func(f *flag.Flag) {
+	fs.VisitAll(func(f *flag.Flag) {
```

Comment:
Not looking into the package, but this like this is just another way to do the same thing. Is there a reason this change was made?

Reply:
It's not the same way, we need that flagset so that we can add our own internal flags to the call; I don't think there is another way to do this than the way that we are doing it here